stdlib/jdbc/src/main/java/org/ballerinax/jdbc/table/SQLDataIterator.java

```diff
@@ -242,7 +242,15 @@ public String getBlob(int columnIndex) {
 
     private void validateAndSetRefRecordField(MapValue<String, Object> bStruct, String fieldName, int expectedTypeTag,
             int actualTypeTag, Object value, String exceptionMessage) throws PanickingApplicationException {
-        setMatchingRefRecordField(bStruct, fieldName, value, expectedTypeTag == actualTypeTag, exceptionMessage);
+        boolean typeMatches = false;
+        if (expectedTypeTag == actualTypeTag) {
+            typeMatches = true;
+        } else {
+            if (actualTypeTag == TypeTags.INT_TAG && expectedTypeTag == TypeTags.BOOLEAN_TAG) {
+                typeMatches = true;
+            }
+        }
```

Comment:
We have a separate method to handle a scenario where there is a set of types for a given actual type to match against.
```java
private void validateAndSetRefRecordField(MapValue<String, Object> bStruct, String fieldName,
            int[] expectedTypeTags, int actualTypeTag, Object value, String exceptionMessage)
```

I think it would be better to use this rather than adding custom logic to this method.